msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/AcquireTokenSilentSupplier.java

```diff
@@ -13,6 +13,8 @@
 class AcquireTokenSilentSupplier extends AuthenticationResultSupplier {
 
     private SilentRequest silentRequest;
+    private boolean shouldRefresh;
+    private boolean afterRefreshOn;
```

Comment:
I think you can avoid adding this boolean to the object state by checking the telemetry flag is set to `REFRESH_REFRESH_IN`

Reply:
Because the telemetry flag is stored as an int several layers into the object passed into this class, I'd have to replace the use of `afterRefreshOn` with the full `clientApplication.serviceBundle().getServerSideTelemetry().getCurrentRequest().cacheInfo() == CacheTelemetry.REFRESH_REFRESH_IN.telemetryValue`

This makes one of the if statements where it's used messier and the other one cleaner, and since that evens out well enough I just got rid of `afterRefreshOn` in the latest commit.